lib/registration.pm

```diff
@@ -239,7 +239,7 @@ sub process_scc_register_addons {
     #    we - Workstation
     #   wsm - Web and Scripting Module
     if (get_var('SCC_ADDONS')) {
-        if (check_screen('scc-beta-filter-checkbox', 5)) {
+        if (check_screen('scc-beta-filter-checkbox', 15)) {
```

Comment:
I'm here with @jknphy

Reply:
Well, probably we could just make that an `assert_screen` based on product version and the BETA variable as well. I also understand that it might not be that easy to understand what to assert for in the alternative case. However, just bumping the timeout is a recipe for failure, just making the issue *more* sporadic and we all know how much we like these cases, right? ;) In https://openqa.suse.de/tests/2375348#step/scc_registration/10 we could look for an empty section below the header of "Available Extensions and Modules". For example see the old QAM SLE12 test https://openqa.suse.de/tests/1860127#step/scc_registration/10 not showing the checkbox. So just create a needle covering the header and the bigger checkbox for the first module to assert that there is no "beta filter checkbox" in between.

Actually I just looked and we already *have* the alternatives since 420199993 so I would really recommend to go with that.

Reply:
Last time I checked this method when I created verification for preselected modules the logic it really looked to me very cumbersome in this function covering all products. I am not sure it has an easy fix with a smart needle which will not break again the test in other product. I appreciate your input but  I think we should merge it to unblock the whole column of the architecture, anyway we are not introducing any delay that was not already there (before I moved the pre-selected check) and we can think more deeply what needs to be done. WDYT?

Reply:
I've noticed that we have `scc-without-beta-filter-checkbox` which we could use, as these 2 needles cover all possible outcomes. As for alternative we could actually use wait_still_screen here to assure that UI is loaded and then it's safe to run `check_screen` with 0 timeout.

Reply:
Would `wait_still_screen` not be the same except that it wastes 7 seconds at least in *all* the cases? I would go for `scc-without-beta-filter-checkbox` same as we do in the other step. This is what I meant with my suggestion above. Sorry if that was not clear.

Reply:
I was taking a closer look to the code:
It fails [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L462) because now we don't execute right after `verify_preselected_modules` (where there is 30s of waiting, which I believe to produce/help the shortcut to hit the UI). In turn we go through [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L242) immediately entering and sending alt-i again (as it is showed in the video that automation uncheck and check back again) when in this point in previous build should be just [waiting](https://openqa.suse.de/tests/2340705#step/scc_registration/11) and not going in that section.
Therefore I think it should help to introduce a waiting time with a needle after this first `alt-i`, meaning checking that we really uncheck after sending the key.